File: babydata/src/main/java/com/zero/babydata/room/ChildDailyRecordDao.java
package com.zero.babydata.room;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.zero.babydata.entity.ChildDailyRecord;

import java.util.List;

@Dao
public interface ChildDailyRecordDao {

    @Insert
    void insertChildDailyRecord(ChildDailyRecord record);


    @Update
    void updateChildDailyRecord(ChildDailyRecord record);

    @Query("SELECT * FROM ChildDailyRecord WHERE babyId = :babyId AND recordDate = :recordDate LIMIT 1")
    ChildDailyRecord getRecordByDate(int babyId, Long recordDate);  // 查询某天记录

    @Query("UPDATE ChildDailyRecord SET weight = :weight WHERE babyId = :babyId AND recordDate = :recordDate")
    void updateWeight(int babyId, Long recordDate, float weight);  // 仅更新体重

    @Query("UPDATE ChildDailyRecord SET height = :height WHERE babyId = :babyId AND recordDate = :recordDate")
    void updateHeight(int babyId, Long recordDate, float height);  // 仅更新身高

    @Query("UPDATE ChildDailyRecord SET pic = :pic WHERE babyId = :babyId AND recordDate = :recordDate")
    void updatePic(int babyId, Long recordDate, String pic);  // 仅更新 pic

    @Query("UPDATE ChildDailyRecord SET extra = :extra WHERE babyId = :babyId AND recordDate = :recordDate")
    void updateExtra(int babyId, Long recordDate, String extra);  // 仅更新 extra

    @Query("SELECT * FROM ChildDailyRecord WHERE babyId = :babyId ORDER BY recordDate DESC")
    LiveData<List<ChildDailyRecord>> getAllRecordsByBabyId(int babyId);
}

File: babydata/src/main/java/com/zero/babydata/room/ChildDailyRecordDao.kt
package com.zero.babydata.room

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update
import com.zero.babydata.entity.ChildDailyRecord

@Dao
interface ChildDailyRecordDao {

    @Insert
    fun insertChildDailyRecord(record: ChildDailyRecord)

    @Update
    fun updateChildDailyRecord(record: ChildDailyRecord)

    @Delete
    fun deleteChildDailyRecord(record: ChildDailyRecord)

    @Query("SELECT * FROM ChildDailyRecord WHERE babyId = :babyId AND recordDate = :recordDate LIMIT 1")
    fun getRecordByDate(babyId: Int, recordDate: Long): ChildDailyRecord?

    @Query("UPDATE ChildDailyRecord SET weight = :weight WHERE babyId = :babyId AND recordDate = :recordDate")
    fun updateWeight(babyId: Int, recordDate: Long, weight: Float)

    @Query("UPDATE ChildDailyRecord SET height = :height WHERE babyId = :babyId AND recordDate = :recordDate")
    fun updateHeight(babyId: Int, recordDate: Long, height: Float)

    @Query("UPDATE ChildDailyRecord SET headCircumference = :headCircumference WHERE babyId = :babyId AND recordDate = :recordDate")
    fun updateHeadCircumference(babyId: Int, recordDate: Long, headCircumference: Float)

    @Query("UPDATE ChildDailyRecord SET pic = :pic WHERE babyId = :babyId AND recordDate = :recordDate")
    fun updatePic(babyId: Int, recordDate: Long, pic: String?)

    @Query("UPDATE ChildDailyRecord SET extra = :extra WHERE babyId = :babyId AND recordDate = :recordDate")
    fun updateExtra(babyId: Int, recordDate: Long, extra: String?)

    @Query("SELECT * FROM ChildDailyRecord WHERE babyId = :babyId ORDER BY recordDate DESC")
    fun getAllRecordsByBabyId(babyId: Int): LiveData<List<ChildDailyRecord>>

    /**
     * 同步查询成长记录（备份使用）
     */
    @Query("SELECT * FROM ChildDailyRecord WHERE babyId = :babyId ORDER BY recordDate DESC")
    fun getAllRecordsByBabyIdSync(babyId: Int): List<ChildDailyRecord>

    @Query("DELETE FROM ChildDailyRecord")
    fun deleteAllChildDailyRecords()
}
